bot/exts/utilities/bookmark.py

```diff
@@ -42,51 +81,59 @@ def build_bookmark_dm(target_message: discord.Message, title: str) -> discord.Em
         return embed
 
     @staticmethod
-    def build_error_embed(user: discord.Member) -> discord.Embed:
-        """Builds an error embed for when a bookmark requester has DMs disabled."""
+    def build_error_embed(message: str) -> discord.Embed:
+        """Builds an error embed for a given message."""
         return discord.Embed(
             title=random.choice(ERROR_REPLIES),
-            description=f"{user.mention}, please enable your DMs to receive the bookmark.",
+            description=message,
             colour=Colours.soft_red
         )
 
+    async def maybe_send_bookmark(
+        self,
+        target_message: discord.Message,
+        title: str,
+        member: discord.Member
+    ) -> discord.Message:
+        """
+        Send the bookmark to the member with the given embed.
+
+        Raise error if they have already bookmarked.
+        Return the DM message object if not.
+        """
+        single_members_bookmarked_messages = await self.get_member_bookmarked_messages(member)
+
+        if target_message.id in single_members_bookmarked_messages:
+            raise AlreadyBookmarkedError
+
+        embed = self.build_bookmark_dm(target_message, title)
+        message = await member.send(embed=embed)
```

Comment:
Seems like this will error if the user cannot be dmed, and won't handle it safely.

Reply:
Ah, it catches the error below. Perhaps update the docstring, since it clarifies one of the issues it will raise but not all.